### ISSUE TITLE
Use remote address

### ISSUE THREAD

**Issue by carkod** [alisaifee/limits#168]
Is this module using remote address somewhere similar to flask-limiter](https://github.com/alisaifee/flask-limiter/blob/dbbc6d8d703ba99c863995d4d9cb4f0dadcc3c38/flask_limiter/extension.py#L442).

Does it use anything to get the remote address by default?

**Comment by alisaifee** [comment]
No, the api requires the caller to provide it.

**Comment by carkod** [comment]
> No, the api requires the caller to provide it.

Thanks for the reply. So how do you limit by IP?

**Comment by alisaifee** [comment]
@carkod the scope of the `limits` library is abstract away the implementation details of using different backend storages such as redis, memcached, mongodb for doing either fixed window or moving window rate limiting. What you rate limit by (for example ip, user etc) is left to the consumer of the library (for example Flask-Limiter, Falcon-Limiter etc).

Could you explain your use case a bit more? Is your application not a Flask application (because then ofcourse, Flask-Limiter will not at all work).

**Comment by carkod** [comment]
> @carkod the scope of the `limits` library is abstract away the implementation details of using different backend storages such as redis, memcached, mongodb for doing either fixed window or moving window rate limiting. What you rate limit by (for example ip, user etc) is left to the consumer of the library (for example Flask-Limiter, Falcon-Limiter etc).
> 
> Could you explain your use case a bit more? Is your application not a Flask application (because then ofcourse, Flask-Limiter will not at all work).

Yes, it is a Flask application, but we are using a Python module to intercept views somethiing like this:

```python
# Flask application
from flask import Flask

app = Flask(__name__)
app.add_url_rule("/homepage", view_func=render_homepage)

# Python module
import flask

def render_homepage():
  
    # Handle rate limits here
    ip_remote_address = 

    return flask.render_template(
            f"templates/homepage"
     )

```

**Comment by alisaifee** [comment]
Since this is related to Flask-Limiter let’s move the issue to that repo
